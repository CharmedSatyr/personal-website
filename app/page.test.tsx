import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Home from "app/page";

describe("Home", () => {
	it("renders a heading", () => {
		render(<Home />);

		const heading = screen.getByRole("heading", { level: 1 });

		expect(heading).toBeInTheDocument();
	});

	it("renders two paragraphs", () => {
		render(<Home />);

		const paragraphs = screen.getAllByRole("paragraph");

		expect(paragraphs.length).toBe(2);
	});

	it("renders two links", () => {
		render(<Home />);

		const links = screen.getAllByRole("link");

		expect(links.length).toBe(2);
	});

	it("should match the last snapshot", () => {
		const { container } = render(<Home />);

		expect(container).toMatchSnapshot();
	});
});
