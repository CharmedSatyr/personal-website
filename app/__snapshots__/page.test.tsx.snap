// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the last snapshot 1`] = `
<div>
  <section
    class="container min-h-96 max-w-3xl"
  >
    <h1
      class="leading-tight"
    >
      I'm a software developer, game designer, and writer in Seattle.
      <img
        alt="tree"
        class="-mt-3 inline"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="45"
        src="/_next/image?url=%2Fstatic%2Fimages%2Ftree-contrast.webp&w=96&q=75"
        srcset="/_next/image?url=%2Fstatic%2Fimages%2Ftree-contrast.webp&w=48&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Ftree-contrast.webp&w=96&q=75 2x"
        style="color: transparent;"
        width="45"
      />
    </h1>
    <p>
      See what I'm up to 
      <a
        href="/now"
      >
        now
      </a>
      .
    </p>
    <p>
      Send me a 
      <a
        href="/contact"
      >
        message
      </a>
      .
    </p>
  </section>
</div>
`;
